pkg/contexts/ocm/transfer/transferhandler/transferhandler.go

```diff
@@ -11,12 +11,65 @@ import (
 	"github.com/open-component-model/ocm/pkg/errors"
 )
 
-type TransferOptions interface{}
+const KIND_TRANSFEROPTION = "transfer option"
 
+// TransferOptions is the general type for an option bag
+// holding options for a transfer handler.
+// Different transfer handler implementations may use differ
+// concrete option sets.
+// Therefore, every option set uses its own options type
+// The options setters/getters MUST be interfaces, which may be implemented
+// by different option set types to enable the option implementations
+// to work with different options sets.
+type TransferOptions interface {
+	NewTransferHandler() (TransferHandler, error)
+}
+
+type TransferOptionsCreator interface {
+	NewOptions() TransferOptions
+}
+
+// TransferOption is an option used to configure a transfer handler.
```

Comment:
Please try and simplify all of this. The bag synonym here does not help the explanation or the usage or the importance of these options that can apply themselves to option bags that then do... what?

Reply:
See above, the transfer handler package is now only for implementors of transfer handlers.

The option handling is explained in detail and is only relevant, if the user wants to provide an own (may be overlapping) set of options.